spark.md

```diff
@@ -167,6 +170,59 @@ Note that ```${scala.binary.version}``` is a Maven property with the value ```2.
 
 The [Deeplearning4j examples repo](https://github.com/deeplearning4j/dl4j-examples) ([old examples here](https://github.com/deeplearning4j/dl4j-spark-cdh5-examples)) contains a number of Spark examples.
 
+## <a name="gpusspark">Training with GPUs on Spark</a>
+
+There are some issues to be aware of when running Deeplearning4j with CUDA on Spark, with cluster managers such as YARN.
+
+
+### <a name="yarngpus">YARN and GPUs</a>
+
+[YARN](https://hadoop.apache.org/docs/current/hadoop-yarn/hadoop-yarn-site/YARN.html) is a commonly used cluster resource management/scheduling tool for Hadoop clusters. Running Deeplearning4j with GPUs on Spark+YARN is possible, but (unlike memory and CPU resources) YARN does not track/handle GPUs as a resource. Consequently, some additional steps are typically required to run DL4J with GPUs on YARN. This is problematic for two reasons:
+
+1. On heterogeneous-hardware clusters (i.e., some machines with GPUs, others without), as YARN won't natively allocate your jobs only to machines that have GPUs
+2. YARN may try to schedule two or more GPU-utilizing jobs to a single machine/GPU
+
+A workaround to this is to utilize node labels (which are available with YARN versions 2.6 or greater). Node labels provide a way to tag or group machines based on your own criteria - in this case, the presence or absence of GPUs. After enabling node labels in the YARN configuration, node labels can be created, and then can be assigned to machines in the cluster. Spark jobs can then be limited to using GPU-containing machines only, by specifying a node label.
+
+Some resources on node labels - and how to configure them - can be found [here](https://hadoop.apache.org/docs/r2.7.3/hadoop-yarn/hadoop-yarn-site/NodeLabel.html) and [here](https://docs.hortonworks.com/HDPDocuments/HDP2/HDP-2.4.2/bk_yarn_resource_mgt/content/configuring_node_labels.html).
```

Comment:
This also works at introductory material:
https://developer.ibm.com/hadoop/2017/03/10/yarn-node-labels/